src/gp_engine.cpp

```diff
@@ -571,17 +641,58 @@ double GPEngine::LogRescalingFor(size_t plv_idx) {
   return static_cast<double>(rescaling_counts_(plv_idx)) * log_rescaling_threshold_;
 }
 
+void GPEngine::SetOptimizationMethod(const GPEngine::OptimizationMethod method) {
+  optimization_method_ = method;
+}
+
+void GPEngine::Optimization(const GPOperations::OptimizeBranchLength& op) {
+  return Optimization(op, optimization_method_);
+}
+
+void GPEngine::Optimization(const GPOperations::OptimizeBranchLength& op,
+                            std::optional<GPEngine::OptimizationMethod> os) {
+  Assert(os.has_value(),
+         "GPEngine::Optimization(): Optimization method has not been set.");
+
+  switch (os.value()) {
```

Comment:
I think making the optimization method `std::optional` was from me, before we added the optimization method assignment to the constructor and removed the default method from enum.  But now I think we can probably just remove `std::optional` from these functions and the `optimization_method_` variable.